### PR TITLE
Add support to validate links in markdown files

### PR DESCRIPTION
**Pull request by rapphil**
**Description:** Add support to validate links in markdown files

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/PR-build.yml [hunk 1]
```diff
@@ -36,8 +36,34 @@ permissions:
...
+          markdown-link-check \
+            --verbose \
+            --config .github/config/markdown-links-config.json \
+            ${{ env.MDS }} \
+            || { echo "Check that anchor links are lowercase"; exit 1; }
+        if: ${{ env.MDS }}
+  
   create-test-ref:
     runs-on: ubuntu-latest
+    needs: validate-markdown
```
Comment (bryan-aguilar) [inline .github/workflows/PR-build.yml:66]:
```suggestion
```
I don't think there needs to be a dependency between these two. Validating markdown can execute independently.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -61,8 +61,33 @@ permissions:
   contents: read
 
 jobs:
+  validate-markdown:
```
[the remark this answers is not in the document]
Comment (bryan-aguilar) [reply .github/workflows/CI.yml:64]: If we do think that this job needs to run at both PR Time and Push time we should separate it into it's own workflow file and give it those workflow triggers.

[the remark this answers is not in the document]
Comment (rapphil) [reply .github/workflows/CI.yml:64]: In pr-build we only validate the files that changed.
in the ci we validate the main md files -> README.md, docs/developer/*.md

Reason why we validate in teh CI as well, is that links might have become invalid. I.e.: the path was deleted in target.

[the remark this answers is not in the document]
Comment (bryan-aguilar) [reply .github/workflows/CI.yml:64]: How expensive would it be to just validate all markdown files at PR Time? It seems like that implementation would save iterations in the case that the CI fails for markdown checks after merge.

[the remark this answers is not in the document]
Comment (bryan-aguilar) [reply .github/workflows/CI.yml:64]: Another plus for it being in it's own workflow is that I don't think this should necessarily block the CI. This should be independent.

[the remark this answers is not in the document]
Comment (rapphil) [reply .github/workflows/CI.yml:64]: we have a lot of md files for previous releases. Validating all of those takes time. (More than 5 minutes If I recall correctly).

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**